### PR TITLE
Fix bug where the legacy storage pool kept being added to the status.

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
BugFix: Legacy CR would cause the storage pool status to be append constantly instead of replaced.
```

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -2738,7 +2738,7 @@ spec:
               name: "metrics"
               protocol: "TCP"
           # Replace this with the built image name
-          image: quay.io/kubevirt/hostpath-provisioner-operator:latest
+          image: registry:5000/hostpath-provisioner-operator:latest
```
Comment (akalenyu) [inline deploy/operator.yaml:2741]: why change? local testing? going to introduce issue in #169?

Comment (awels) [reply deploy/operator.yaml]: local testing, I didn't mean to check in the file.

**Pull request merged**